plugins/gc/common/src/main/java/org/qbicc/plugin/gc/common/safepoint/SafePoints.java

```diff
@@ -33,31 +25,15 @@ public static void enqueueMethods(CompilationContext ctxt) {
      * @return the basic block builder (not {@code null})
      */
     public static BasicBlockBuilder createBasicBlockBuilder(BasicBlockBuilder.FactoryContext fc, BasicBlockBuilder delegate) {
-        final AbstractSafePointStrategy strategy = delegate.getContext().getAttachment(IMPL_KEY);
+        final ClassContext bcc = delegate.getContext().getBootstrapClassContext();
+        final DefinedTypeDefinition dt = bcc.findDefinedType(THREAD_INT_NAME);
         return new DelegatingBasicBlockBuilder(delegate) {
+
             @Override
             public Node safePoint() {
-                strategy.safePoint(getFirstBuilder());
-                return nop();
+                final MethodElement pollSafePoint = dt.load().requireSingleMethod("pollSafePoint");
+                return call(getLiteralFactory().literalOf(pollSafePoint), List.of());
```

Comment:
This requires that `pollSafePoint` exists on `Thread`, so it'll likely require another classlibs release. Need to think about ordering a bit.